bin/prettierd.js

```diff
@@ -1,9 +1,22 @@
 #!/usr/bin/env node
 
+const path = require("path");
+const fs = require("fs");
+
 process.env.CORE_D_TITLE = "prettierd";
 process.env.CORE_D_DOTFILE = ".prettierd";
 process.env.CORE_D_SERVICE = require.resolve("../dist/service");
 
+if (process.env.HOME && process.env.XDG_CONFIG_HOME) {
+  const confDir = path.join(
+    process.env.XDG_CONFIG_HOME,
+    process.env.CORE_D_TITLE
+  );
+  const confPath = path.join(confDir, process.env.CORE_D_TITLE);
+  fs.mkdir(confDir, { recursive: true }, () => {});
+  process.env.CORE_D_DOTFILE = path.relative(process.env.HOME, confPath);
+}
```

Comment:
This assumes `XDG_CONFIG_HOME` is under `HOME`. Should we validate that or is it required that `XDG_CONFIG_HOME` is always going to be a subdirectory of `HOME`?

Reply:
it seems cored is expecting CORE_D_DOTFILE to be relative to HOME (and abs paths don't work either)
so in the case where XDG_CONFIG_HOME is not a child of HOME, `path.relative` returns something like `../../home/senan/.config/...` which works as expected